cli/delete.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	datapb "go.viam.com/api/app/data/v1"
+)
+
+// DeleteBinaryData deletes binary data matching filter.
+func (c *AppClient) DeleteBinaryData(dst string, filter *datapb.Filter) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+	resp, err := c.sendDeleteBinaryDataByFilterRequest(filter)
+	if err != nil {
+		return errors.Wrapf(err, "received error from server")
+	}
+	deletedCount := resp.GetDeletedCount()
+	status := resp.GetResult().GetStatus()
+
+	switch status {
+	case datapb.Status_STATUS_PARTIAL_SUCCESS:
+		fmt.Fprint(c.c.App.Writer, "errors", resp.GetResult().GetMessage())
+	case datapb.Status_STATUS_SUCCESS:
+		fmt.Fprintf(c.c.App.Writer, "deleted %d files", deletedCount)
+	default:
+		break
+	}
+
+	return nil
+}
+
+func (c *AppClient) sendDeleteBinaryDataByFilterRequest(filter *datapb.Filter) (*datapb.DeleteBinaryDataByFilterResponse, error) {
```

Comment:
Since these two helper functions aren't adding additional logic and are quite short as-is, worth keeping the requests directly inline with the code above. Gives the benefit of readability above without having to redirect to what the helper functions do.

Reply:
done!